### ISSUE TITLE
Stack trace encountered when staging a tarball

### ISSUE THREAD

**Issue by Cynical-Optimist** [buildstream-migration/bst-staging#155]
[See original issue on GitLab](https://gitlab.com/BuildStream/buildstream/-/issues/155)
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Nov 15, 2017, 09:22

So far this is an incident, not sure how to reproduce:

```
[00:01:15][67f528b9][build:core/adwaita-icon-theme.bst   ] BUG     gnome/core-adwaita-icon-theme/67f528b9-build.15748.log

    An unhandled exception occured:
    
    Traceback (most recent call last):
      File "/codethink/GNOME/buildstream/buildstream/_scheduler/job.py", line 265, in child_action
        result = self.action(element)
      File "/codethink/GNOME/buildstream/buildstream/_scheduler/buildqueue.py", line 34, in process
        element._assemble()
      File "/codethink/GNOME/buildstream/buildstream/element.py", line 1074, in _assemble
        self.stage(sandbox)
      File "/codethink/GNOME/buildstream/buildstream/buildelement.py", line 177, in stage
        self.stage_sources(sandbox, self.get_variable('build-root'))
      File "/codethink/GNOME/buildstream/buildstream/element.py", line 481, in stage_sources
        self._stage_sources_at(host_directory)
      File "/codethink/GNOME/buildstream/buildstream/element.py", line 1406, in _stage_sources_at
        source._stage(directory)
      File "/codethink/GNOME/buildstream/buildstream/source.py", line 316, in _stage
        self.stage(directory)
      File "/codethink/GNOME/buildstream/buildstream/plugins/sources/tar.py", line 78, in stage
        tar.extractall(path=directory, members=self._extract_members(tar, base_dir))
      File "/usr/lib/python3.5/tarfile.py", line 1998, in extractall
        numeric_owner=numeric_owner)
      File "/usr/lib/python3.5/tarfile.py", line 2040, in extract
        numeric_owner=numeric_owner)
      File "/usr/lib/python3.5/tarfile.py", line 2118, in _extract_member
        self.makelink(tarinfo, targetpath)
      File "/usr/lib/python3.5/tarfile.py", line 2205, in makelink
        self._extract_member(self._find_link_target(tarinfo),
      File "/usr/lib/python3.5/tarfile.py", line 2372, in _find_link_target
        raise KeyError("linkname %r not found" % linkname)
    KeyError: "linkname 'adwaita-icon-theme-3.26.0/Adwaita/cursors/6407b0e94181790501fd1e167b474872' not found"
```

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Nov 15, 2017, 09:42

100% reproducible.

This appears to happen whenever trying to stage this tarball:
    https://download.gnome.org/sources/adwaita-icon-theme/3.26/adwaita-icon-theme-3.26.0.tar.xz

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Nov 17, 2017, 08:34

mentioned in commit 1e9207723ad5ad56731bd23eb696dfaf6e8af593

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @tristanvb]](https://gitlab.com/tristanvb) on Nov 17, 2017, 08:34

closed via commit 28328cde18b14677030235baecea29e38900f75f

**Comment by Cynical-Optimist** [comment]
In GitLab by [[Gitlab user @jonathanmaw]](https://gitlab.com/jonathanmaw) on Nov 17, 2017, 09:52

mentioned in commit 28328cde18b14677030235baecea29e38900f75f
